assets/js/influxdb-url.js

```diff
@@ -353,26 +390,48 @@ showPreference()
 
 // Validate custom URLs
 function validateUrl(url) {
-  try {
-    new URL(url);
-    return {valid: true, error: ""}
-  } catch(e) {
-    var validProtocol = /^http(s?)/
-    var protocol = url.match(/http(s?):\/\//) ? url.match(/http(s?):\/\//)[0] : "";
-    var domain = url.replace(protocol, "")
-    /** validDomain = (Named host | IPv6 host | IPvFuture host)(:Port)? **/
-    var validDomain = new RegExp(`([a-z0-9\-._~%]+`
-                               + `|\[[a-f0-9:.]+\]`
-                               + `|\[v[a-f0-9][a-z0-9\-._~%!$&'()*+,;=:]+\])`
-                               + `(:[0-9]+)?`);
-    if (validProtocol.test(protocol) == false) {
-      return {valid: false, error: "Invalid protocol, use http[s]"}
+  /** validDomain = (Named host | IPv6 host | IPvFuture host)(:Port)? **/
+  var validDomain = new RegExp(`([a-z0-9\-._~%]+`
+  + `|\[[a-f0-9:.]+\]`
+  + `|\[v[a-f0-9][a-z0-9\-._~%!$&'()*+,;=:]+\])`
+  + `(:[0-9]+)?`);
+
+  if (context() !== 'dedicated') {
+    // Validation for non-dedicated custom InfluxDB URLs
+    try {
+      new URL(url);
+      return {valid: true, error: ""}
+    } catch(e) {
+      if (context() !== 'dedicated') {
```

Comment:
Oops, forgot to take that out. Thanks! I had the conditional fork here at first, but moved up before `try` because dedicated URLs shouldn't include the protocol so `new URL(url)` was always throwing an error.